test/models/exercise_map_test.rb

```diff
@@ -15,6 +15,8 @@ class ExerciseMapTest < ActiveSupport::TestCase
     unlocked_exercise = create(:exercise,
                                unlocked_by: exercise,
                                track: track)
+    create(:exercise, track: track, unlocked_by: create(:exercise))
+    solution = create(:solution, exercise: exercise)
```

Comment:
Could you add a comment here to explain why these two lines are necessary else I can see them getting refactored out later and us regressing.